packages/cw-derive/examples/basic.rs

```diff
@@ -0,0 +1,30 @@
+#![allow(dead_code)]
+use cosmwasm_std::Response;
+
+struct Ctx;
+struct Error;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, schemars::JsonSchema)]
+struct Member;
+
+#[cw_derive::interface(module=msg, exec=Execute, query=Query)]
+trait Cw4 {
+    #[msg(exec)]
+    fn update_admin(&self, ctx: Ctx, admin: Option<String>) -> Result<Response, Error>;
+
+    #[msg(exec)]
```

Comment:
Yes, `cw_derive::interface` is basically scanning for those `msg` attributes to generate inferface.

There is a reason not to hardcode the names - becase `Cw1Exec` or `Cw1ExecMsg` is way better than `ExecMsg` which is default. It could be reasonable to have default as `TraitNameExecMsg`/`TraitNameQueryMsg`, but it also may not always work. Making it configurable with some default is now way easier now than complicating generation later.

Reply:
Fair enough.
As I guessed, you were 2 steps ahead of me.